pydantic_ai_slim/pydantic_ai/models/bedrock.py

```diff
@@ -393,7 +405,11 @@ async def _map_messages(
                                     {
                                         'toolResult': {
                                             'toolUseId': part.tool_call_id,
-                                            'content': [{'text': part.model_response_str()}],
+                                            'content': [
+                                                {'text': part.model_response_str()}
+                                                if profile.bedrock_tool_result_format == 'text'
+                                                else {'json': part.model_response_object()}
+                                            ],
```

Comment:
See https://github.com/pydantic/pydantic-ai/issues/1649#issuecomment-2856042380, here to verify model-specific profile settings work